### ISSUE TITLE
apparmor logs flood /var/log/syslog

### ISSUE THREAD

**Issue by padcom** [docker/docker#19284]
My ```/var/log/syslog``` is flooded with messages from apparmor regarding docker:

```
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149377] audit: type=1400 audit(1452649444.219:6421): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149412] audit: type=1400 audit(1452649444.219:6422): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149428] audit: type=1400 audit(1452649444.219:6423): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149442] audit: type=1400 audit(1452649444.219:6424): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149460] audit: type=1400 audit(1452649444.219:6425): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149474] audit: type=1400 audit(1452649444.219:6426): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149486] audit: type=1400 audit(1452649444.219:6427): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149500] audit: type=1400 audit(1452649444.219:6428): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:04 PL-KRA-MOB-109L kernel: [ 1761.149512] audit: type=1400 audit(1452649444.219:6429): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
Jan 13 02:44:07 PL-KRA-MOB-109L kernel: [ 1764.155968] audit: type=1400 audit(1452649447.219:6430): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
```

I get a piece like that every 7 seconds for every container on every machine running docker.

In my opinion it is the same as #7276 and others that have been closed months (even years) ago but the problem has never really been fixed.

Running Docker version 1.9.1, build a34a1d5

**Comment by padcom** [comment]
With all due respect I'd not report a duplicate (which clearly this bug report is as denoted by the reference to #7276) had the original issue been fixed (which it isn't) or left open (I'd add a comment there).

To signify: #7276 has been opened on Jul 28, 2014, closed Jan 6, 2015 with no code changes that would actually fix it (just another issue related to apparmor profile) and since then it is bluntly ignored that the issue is still there producing loads of logs making it impossible to have the syslog readable.

Don't get me wrong: I love docker as a tool but things like that make Docker's opponents points very much valid and it is hard to convince people otherwise.

**Comment by thaJeztah** [comment]
What is running in those containers? IIRC  `ptrace` was blocked by default, but you can override that https://github.com/docker/docker/issues/7276#issuecomment-68812214.

Also see https://github.com/docker/docker/pull/18393, which may resolve this for you, and will be in docker 1.10

**Comment by padcom** [comment]
samba, Apache web server, transmission, GitLab, Oracle,... doesn't matter. It happens with each and every container regardless of its content.

**Comment by thaJeztah** [comment]
@padcom is this still an issue for you on docker 1.11?

**Comment by thaJeztah** [comment]
This should be resolved in 1.11 through https://github.com/docker/docker/pull/20305 and https://github.com/docker/docker/pull/18393, so I'll go ahead and close it, but let me know if you still see this on 1.11

**Comment by padcom** [comment]
I honestly didn't upgrade yet - I will give it a shot and let you know

**Comment by jeffbyrnes** [comment]
This fixed our apparmor throwing the `ptrace` errors to our syslog.
